### PR TITLE
Add Slider to AntiAFK for Custom Movement Increment Setting

### PR DESCRIPTION
**Pull request by Developer78-sgyuijhgygwtdwgyhutre45r5t**
## References
See #280

## Summary by CodeRabbit

- **New Features**
	- Added customizable settings for Anti-AFK timing using sliders, allowing users to set specific intervals between movements.
- **Enhancements**
	- Improved the Anti-AFK feature's timing logic for more dynamic and user-responsive behavior.

### CODE REVIEW

File: src/main/java/net/wurstclient/hacks/AntiAfkHack.java [hunk 1]
```diff
@@ -28,15 +28,23 @@
...
+	private final SliderSetting movetime =
+		new SliderSetting("Time between movements in ticks",
+			"Time between movements in ticks. If set to 0, it will be random.",
+			0, 0, 350, 5,
+			ValueDisplay.DECIMAL.withSuffix(" Ticks").withLabel(0, "Random"));
+	
 	private final CheckboxSetting useAi = new CheckboxSetting("Use AI", true);
 	
-	private int timer;
+	private float timer;
```
Comment (Alexander01998) [inline src/main/java/net/wurstclient/hacks/AntiAfkHack.java:47]: `timer` should be an `int`, not a `float`.

File: src/main/java/net/wurstclient/hacks/AntiAfkHack.java [hunk 2]
```diff
@@ -81,6 +90,17 @@ protected void onDisable()
...
 	}
 	
+	private void updateMovementValue()
+	{
+		if(movetime.getValueF() == 0)
+		{
+			timer = 40 + random.nextInt(21);
+		}else
+		{
+			timer = movetime.getValueF();
```
Comment (Alexander01998) [inline src/main/java/net/wurstclient/hacks/AntiAfkHack.java:100]: This needs to be `getValueI()`.

File: src/main/java/net/wurstclient/hacks/AntiAfkHack.java [hunk 3]
```diff
@@ -138,32 +151,28 @@ public void onUpdate()
...
 			// walk
 			if(MC.player.squaredDistanceTo(Vec3d.ofCenter(nextBlock)) > 0.5)
 				MC.options.forwardKey.setPressed(true);
 			else
 				MC.options.forwardKey.setPressed(false);
-			
 			// swim up
 			MC.options.jumpKey.setPressed(MC.player.isTouchingWater());
 			
-			// update timer
```
Comment (Alexander01998) [inline src/main/java/net/wurstclient/hacks/AntiAfkHack.java:166]: There are quite a few seemingly random formatting changes and removed comments here. Please revert.

File: src/main/java/net/wurstclient/hacks/AntiAfkHack.java [hunk 4]
```diff
@@ -81,6 +90,17 @@ protected void onDisable()
 		PathProcessor.releaseControls();
 	}
 	
+	private void updateMovementValue()
```
Comment (Alexander01998) [inline src/main/java/net/wurstclient/hacks/AntiAfkHack.java:93]: Consider giving this method a more descriptive name, like `setTimer()`.

File: src/main/java/net/wurstclient/hacks/AntiAfkHack.java [hunk 5]
```diff
@@ -28,15 +28,23 @@
...
 @SearchTags({"anti afk", "AFKBot", "afk bot"})
 @DontSaveState
 public final class AntiAfkHack extends Hack
 	implements UpdateListener, RenderListener
 {
+	private final SliderSetting movetime =
+		new SliderSetting("Time between movements in ticks",
+			"Time between movements in ticks. If set to 0, it will be random.",
+			0, 0, 350, 5,
+			ValueDisplay.DECIMAL.withSuffix(" Ticks").withLabel(0, "Random"));
```
Comment (Alexander01998) [inline src/main/java/net/wurstclient/hacks/AntiAfkHack.java:43]: For consistency with other settings, slider suffixes and labels should always be lowercase.

BTW @coderabbitai remember this: The `withPrefix()`, `withSuffix()`, and `withLabel()` methods of `net.wurstclient.settings.SliderSetting.ValueDisplay` should be called with a lowercase String.

**Review by Alexander01998 — CHANGES_REQUESTED** [review]

**Comment by Developer78-sgyuijhgygwtdwgyhutre45r5t** [comment]
Hello @Alexander01998, Thanks for taking the time to review my pull request. I have made the requested changes. If there are any further changes you would like me to make, feel free to let me know.

**Review by Alexander01998 — APPROVED** [review]

File: src/main/java/net/wurstclient/hacks/AntiAfkHack.java [hunk 6]
```diff
@@ -81,6 +90,14 @@ protected void onDisable()
 		PathProcessor.releaseControls();
 	}
 	
+	private void setTimer()
+	{
+		if(movetime.getValueI() == 0)
+			timer = 40 + random.nextInt(21);
+		else
+			timer = movetime.getValueI();
```
[the remark this answers is not in the document]
Comment (Alexander01998) [reply src/main/java/net/wurstclient/hacks/AntiAfkHack.java:98]: No.

File: src/main/java/net/wurstclient/hacks/AntiAfkHack.java [hunk 7]
```diff
@@ -81,6 +96,15 @@ protected void onDisable()
...
 	}
 	
+	private void setTimer()
+	{
+		int baseTime = (int)(waitTime.getValue() * 20);
+		int randTime = (int)(waitTimeRand.getValue() * 20);
+		int randOffset = random.nextInt(randTime * 2 + 1) - randTime;
+		randOffset = Math.max(randOffset, -baseTime);
+		timer = baseTime + randOffset;
+	}
```
[the remark this answers is not in the document]
Comment (Alexander01998) [reply src/main/java/net/wurstclient/hacks/AntiAfkHack.java:106]: Still no.

**Review by JamesWalton69 — APPROVED** [review]

**Pull request merged**